girder/api/v1/notification.py

```diff
@@ -112,22 +113,25 @@ def streamGen():
     @autoDescribeRoute(
         Description('List notification events')
         .notes('This endpoint can be used for manual long-polling when '
-               'SSE support is disabled or otherwise unavailable.')
-        .param('since', 'Filter out events before this time stamp.',
-               dataType='integer', required=False)
+               'SSE support is disabled or otherwise unavailable. The events are always '
+               'returned in chronological order.')
+        .param('since', 'Filter out events before this date. Pass as either epoch seconds or '
+               'an ISO-8601 formatted datetime.', required=False)
         .errorResponse()
         .errorResponse('You are not logged in.', 403)
     )
     def listNotifications(self, since):
         user, token = self.getCurrentUser(returnToken=True)
-        timestamp = time.time()
 
         if since is not None:
-            since = datetime.utcfromtimestamp(since)
-        notifications = list(NotificationModel().get(user, since, token=token))
-
-        if notifications:
-            timestamp = max(n['updatedTime'] for n in notifications)
-
-        setResponseHeader('Date', str(timestamp))
-        return notifications
+            try:
+                since = dateutil.parser.parse(since)
+            except ValueError:
+                try:
+                    since = datetime.utcfromtimestamp(float(since))
```

Comment:
All other parts of Girder serialize and expect to parse times in ISO-8601. The most source of the `since` parameter will be if an application extracts the most recent `updated` (which is returned as ISO-8601) from an old list of notifications.

Javascript can also easily create ISO-8601 representations of current or arbitrary times: [`new Date().toISOString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString).

So, why are we supporting epoch seconds in this one place (but not the rest of Girder) at all? I'd vote to simplify and expect everything as ISO format.

Reply:
Fine by me.